app/src/main/java/org/fossasia/openevent/general/speakercall/SpeakersCallViewModel.kt

```diff
@@ -6,30 +6,91 @@ import androidx.lifecycle.ViewModel
+import org.fossasia.openevent.general.auth.AuthService
+import org.fossasia.openevent.general.auth.User
 import org.fossasia.openevent.general.common.SingleLiveEvent
 import org.fossasia.openevent.general.data.Resource
 import org.fossasia.openevent.general.event.EventService
+import org.fossasia.openevent.general.speakers.Speaker
+import org.fossasia.openevent.general.speakers.SpeakerService
 import org.fossasia.openevent.general.utils.extensions.withDefaultSchedulers
 import timber.log.Timber
 
 class SpeakersCallViewModel(
     private val eventService: EventService,
-    private val resource: Resource
+    private val resource: Resource,
+    private val authHolder: AuthHolder,
+    private val authService: AuthService,
+    private val speakerService: SpeakerService
 ) : ViewModel() {
 
     private val compositeDisposable = CompositeDisposable()
 
     private val mutableSpeakersCall = MutableLiveData<SpeakersCall>()
     val speakersCall: LiveData<SpeakersCall> = mutableSpeakersCall
-    private val mutableError = SingleLiveEvent<String>()
-    val errorMessage: LiveData<String> = mutableError
+    private val mutableMessage = SingleLiveEvent<String>()
+    val message: LiveData<String> = mutableMessage
     private val mutableProgress = MutableLiveData<Boolean>(true)
     val progress: LiveData<Boolean> = mutableProgress
+    private val mutableSpeaker = MutableLiveData<Speaker>()
+    val speaker: LiveData<Speaker> = mutableSpeaker
+    private val mutableUser = MutableLiveData<User>()
+    val user: LiveData<User> = mutableUser
+    private val mutableEmptySpeakersCall = MutableLiveData<Boolean>(true)
+    val emptySpeakersCall: LiveData<Boolean> = mutableEmptySpeakersCall
+
+    fun isLoggedIn(): Boolean = authHolder.isLoggedIn()
+
+    fun loadMyUserAndSpeaker(eventId: Long, eventIdentifier: String) {
+        compositeDisposable += authService.getProfile()
+            .withDefaultSchedulers()
+            .doOnSubscribe {
+                mutableProgress.value = true
+            }.doFinally {
+                mutableProgress.value = false
+            }.subscribe({ user ->
+                mutableUser.value = user
+                loadMySpeaker(user, eventId, eventIdentifier)
+            }) {
+                Timber.e(it, "Failure")
+                mutableMessage.value = resource.getString(R.string.failure)
+            }
+    }
+
+    fun loadMySpeaker(user: User, eventId: Long, eventIdentifier: String) {
+        val query = """[{
+                |   'and':[{
+                |       'name':'event',
+                |       'op':'has',
+                |       'val': {
+                |           'name': 'identifier',
+                |           'op': 'eq',
+                |           'val': '$eventIdentifier'
+                |       }
+                |    }, {
+                |       'name':'email',
+                |       'op':'eq',
+                |       'val':'${user.email}'
```

Comment:
Why is the email thingy needed?

Reply:
I followed the implementation in the frontend. It seems like the server allow a user to create multiple speaker's profiles, but only allow one email per profile. 

<img src="https://i.ibb.co/QnjRZdy/dfs.png" alt="dfs" border="0">